tensorboard/data/experimental/experiment_from_dev_test.py

```diff
@@ -0,0 +1,250 @@
+            yield response
+
+        mock_api_client.StreamExperimentData = mock.Mock(
+            wraps=stream_experiment_data
+        )
+
+        stderr_messages = []
+
+        def fake_stderr_write(msg):
+            stderr_messages.append(msg)
+
+        with mock.patch.object(
+            experiment_from_dev,
+            "get_api_client",
+            lambda api_endpoint: mock_api_client,
+        ):
+            experiment = experiment_from_dev.ExperimentFromDev("789")
+            with mock.patch.object(sys.stderr, "write", fake_stderr_write):
+                dataframe = experiment.get_scalars()
+            self.assertLen(stderr_messages, 1)
+            self.assertRegexpMatches(
+                stderr_messages[0],
+                r"missing value\(s\).*uneven numbers of steps.*pivot=False",
+            )
+
+    def test_get_scalars_with_non_pivot_table(self):
+        mock_api_client = mock.Mock()
+
+        def stream_experiment_data(request, **kwargs):
+            self.assertEqual(request.experiment_id, "789")
+            self.assertEqual(kwargs["metadata"], grpc_util.version_metadata())
+            for run in ("train", "test"):
+                for tag in ("accuracy", "loss"):
+                    response = export_service_pb2.StreamExperimentDataResponse()
+                    response.run_name = run
+                    response.tag_name = tag
+                    display_name = "%s:%s" % (request.experiment_id, tag)
+                    response.tag_metadata.CopyFrom(
+                        test_util.scalar_metadata(display_name)
+                    )
+                    for step in range(10):
+                        response.points.steps.append(step)
+                        if tag == "loss":
+                            if run == "train":
+                                value = 1.0 / (step + 1)
+                                seconds = step
+                            else:
+                                value = -1.0 / (step + 1)
+                                seconds = 600 + step
+                        else:  # "accuracy"
+                            if run == "train":
+                                value = 1.0 / (10 - step)
+                                seconds = step * 2
+                            else:
+                                value = -1.0 / (10 - step)
+                                seconds = 600 + step * 2
+                        response.points.values.append(value)
+                        response.points.wall_times.add(seconds=seconds, nanos=0)
+                    yield response
+
+        mock_api_client.StreamExperimentData = mock.Mock(
+            wraps=stream_experiment_data
+        )
+
+        with mock.patch.object(
+            experiment_from_dev,
+            "get_api_client",
+            lambda api_endpoint: mock_api_client,
+        ):
+            experiment = experiment_from_dev.ExperimentFromDev("789")
```

Comment:
It looks like this giant mock block is identical to the one in the
previous first case. Could you please combine the test cases, using
`self.subTest` for failure isolation?

Reply:
Done.